deploy/stacks/pipeline.py

```diff
@@ -209,14 +136,14 @@ def __init__(
                     build_image=codebuild.LinuxBuildImage.AMAZON_LINUX_2_4,
                 ),
                 commands=[
-                    f'aws codeartifact login --tool npm --repository {self.codeartifact.npm_repo.attr_name} --domain {self.codeartifact.domain.attr_name} --domain-owner {self.codeartifact.domain.attr_owner}',
+                    f'aws codeartifact login --tool npm --repository {self.codeartifact.codeartifact_npm_repo_name} --domain {self.codeartifact.codeartifact_domain_name} --domain-owner {self.codeartifact.domain.attr_owner}',
                     'npm install -g aws-cdk',
-                    f'aws codeartifact login --tool pip --repository {self.codeartifact.pip_repo.attr_name} --domain {self.codeartifact.domain.attr_name} --domain-owner {self.codeartifact.domain.attr_owner}',
+                    f'aws codeartifact login --tool pip --repository {self.codeartifact.codeartifact_pip_repo_name} --domain {self.codeartifact.codeartifact_domain_name} --domain-owner {self.codeartifact.domain.attr_owner}',
                     'pip install -r deploy/requirements.txt',
                     'cdk synth',
                     'echo ${CODEBUILD_SOURCE_VERSION}'
                 ],
-                role_policy_statements=self.codebuild_policy,
+                role=self.pipeline_iam_role,
```

Comment:
This goes back to the previous code where we had the following configuration:

Baseline CodeBuild Policies were assigned to the following CodeBuild Steps:
- Synth
- ALL Quality Gate CodeBuild Steps
- Lambda Image
- ECS Image

Baseline CodeBuild Policies + select sts/cloudfront permissions were assigned to:
- DB Migration
- Stack Updater
- Deploy FrontEnd
- Update Documentation
- Configure RUM
- Configure Cognito
- ALBFrontEndImage
- UserGuideImage

Baseline CodeBuild Policies + sts/cloudfront + select codecommit permissions were assigned to:
- GitRelease

I created 3 roles to align with the above 3 sets of permissions, `pipeline_iam_role`, `codebuild_role`, and `git_project_role` keeping the permissions the same and refactoring to focus on improving management of these permission sets. 

I will rename the roles to be `baseline_codebuild_role`, `expanded_codebuild_role`, and `git_project_role` to avoid confusion with naming. 

I think an optimization here would be to define individual permission sets for each CodeBuild Role (or each logical grouping of CodeBuild Steps) but would opt to leave that as a separate PR unless others think differently?

Reply:
My bad for not looking into each role policies! I see now. And yes and additional improvement would be to define each role/set of roles separatedly. But I agree, this can be part of a different PR